### PR TITLE
Remove the deprecated insecure serving from the cloud controller manager

### LINKED ISSUE #108952: Remove insecure serving configuration from cloud controller manager libraries
The [insecure serving configuration](https://github.com/kubernetes/cloud-provider/blob/v0.24.0-alpha.4/app/config/config.go#L37-L38) in the cloud controller manager libraries is deprecated, and similar configuration has been removed from most other components (apiserver https://github.com/kubernetes/kubernetes/issues/91506, kube controller manager https://github.com/kubernetes/kubernetes/pull/106860).  We should remove the insecure serving flags and configuration from the cloud controller manager libraries to complete the deprecation and be consistent with the kube controller manager.

/sig cloud-provider

### PR DESCRIPTION
**Pull request by nckturner**
#### What type of PR is this?

/kind cleanup
/kind deprecation

#### What this PR does / why we need it:

Remove deprecated "insecure serving" configuration from the cloud-provider package. 

#### Which issue(s) this PR fixes:

Fixes #108952

#### Special notes for your reviewer:

/sig cloud-provider

#### Does this PR introduce a user-facing change?

```release-note
Remove insecure serving configuration from cloud-provider package, which is consumed by cloud-controller-managers.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/cloud-provider/options/options.go [hunk 1]
```diff
@@ -133,7 +123,6 @@ func (o *CloudControllerManagerOptions) Flags(allControllers, disabledByDefaultC
 	o.ServiceController.AddFlags(fss.FlagSet("service controller"))
 
 	o.SecureServing.AddFlags(fss.FlagSet("secure serving"))
-	o.InsecureServing.AddUnqualifiedFlags(fss.FlagSet("insecure serving"))
```
[the remark this answers is not in the document]
Comment (nckturner) [reply staging/src/k8s.io/cloud-provider/options/options.go:136]: Not hidden, but marked as deprecated: 
```
Insecure serving flags:                                                                                                                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                                                     
      --address ip                                                                                                                                                                                                                                                                                                                                                           
                The IP address on which to serve the insecure --port (set to '0.0.0.0' or '::' for listening in all interfaces and IP families). (default 0.0.0.0) (DEPRECATED: see --bind-address instead.)                                                                                                                                                                         
      --port int                                                                                                                                                                                                                                                                                                                                                                     
                The port on which to serve unsecured, unauthenticated access. Set to 0 to disable. (DEPRECATED: see --secure-port instead.)           
```

[the remark this answers is not in the document]
Comment (cheftako) [reply staging/src/k8s.io/cloud-provider/options/options.go:136]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli. Checked the timeline and field has been deprecated for the last 4 years.

**Comment by nckturner** [comment]
We discussed this in sig-cloud-provider today and determined we would like to ask for an exception for code freeze. The change is small and removes insecure serving configuration from cloud-controller-managers which has been deprecated for 4 years.

/triage accepted
/milestone v1.24

**Pull request merged**